### PR TITLE
fix(aws-lambda): handle multiple cookies in streaming responses

### LINKED ISSUE #2921: Setting multiple cookies at once does not work using lambda streamHandle
### What version of Hono are you using?

4.4.3

### What runtime/platform is your app running on?

AWS Lambda

### What steps can reproduce the bug?

Create a response of a request which is setting multiple cookies with `Set-Cookie` header.

### What is the expected behavior?

All cookies are sent on headers to the client response.

### What do you see instead?

Only one header is sent to the client response.

### Additional information

[This line in the code](https://github.com/honojs/hono/blob/main/src/adapter/aws-lambda/handler.ts#L130-L133) does overwrite multiple headers with the same name.

[As documented here](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-lambda.html) the cookies should be returned as an array of strings to the lambda runtime.

If you accept this, I'll provide a PR to fix this.

### PR DESCRIPTION
**Pull request by KnisterPeter**
This fix does address that cookies have to be handled
specially in lambda responses.

Fixes #2921

### CODE REVIEW

**Comment by KnisterPeter** [comment]
It fails in the coverage check because of some GitHub rate-limit. 😿

File: src/adapter/aws-lambda/handler.ts [hunk 1]
```diff
@@ -126,10 +126,21 @@ export const streamHandle = <
...
+            headers[name] = value
+          }
+        })
+
         // Check content type
         const httpResponseMetadata = {
           statusCode: res.status,
-          headers: Object.fromEntries(res.headers.entries()),
+          headers,
+          cookies,
```
Comment (watany-dev) [inline src/adapter/aws-lambda/handler.ts:143]: ok.

https://github.com/jill64/types-lambda/blob/main/types/streaming/ResponseStreamMetadata.ts

**Comment by KnisterPeter** [comment]
@yusukebe @usualoma Can you share an update on this bugfix? Will it be able to get merged? If not what do I need to provide to have it fixed?

**Comment by yusukebe** [comment]
@KnisterPeter 

Sorry for not responding to you! It seems to be good. But if you can, can you write a unit test?

**Comment by yusukebe** [comment]
@watany-dev Do you have any ideas for testing `streamHandle`?

**Comment by KnisterPeter** [comment]
@yusukebe Found a way to test the cookies (and other headers) in the stream. Can you accept that solution?

**Comment by KnisterPeter** [comment]
@yusukebe What else can I do to get this merged?

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@KnisterPeter 

Sorry for my late reply. This is great! I appreciate you added the test. Let's go with it. I'll merge it and release a minor version, including it later. Thanks.

**Comment by KnisterPeter** [comment]
Hi @yusukebe, can you do a minor release with this patch? We are seeing this issue and waiting for a fix.

**Comment by yusukebe** [comment]
@KnisterPeter 

I've released the `4.4.6` which includes this fix now! Thanks!

**Pull request merged**